dvc/repo/metrics/show.py

```diff
@@ -1,14 +1,19 @@
 from __future__ import unicode_literals
 
+import errno
 import os
 import csv
 import json
 import logging
+
 from jsonpath_ng.ext import parse
 
 from dvc.exceptions import OutputNotFoundError, BadMetricError, NoMetricsError
 from dvc.utils.compat import builtin_str, open, StringIO, csv_reader
 
+NO_METRICS_FILE_AT_BRANCH_WARNING = (
+    "Metrics file '{}' does not exist at branch: '{}'"
```

Comment:
Btw, it is a reoccurring issue within this whole file, but it is not always a `branch` that we are getting, it could be a tag as well. Should we generalize it and call it `reference` or something? Would that make sense?

Reply:
It is not really related to your patch, but I've just noticed it and had to mention :slightly_smiling_face:

Reply:
Sure, ill do that.